libs/shared/flagd-core/src/lib/targeting/fractional.ts

```diff
@@ -0,0 +1,97 @@
+import {flagdPropertyKey, flagKeyPropertyKey, targetingPropertyKey} from "./common";
+import MurmurHash3 from "imurmurhash";
+
+export const fractionalRule = "fractional";
+
+export function fractional(data: unknown, context: Record<any, any>): string | null {
+  if (!Array.isArray(data)) {
+    return null;
+  }
+
+  const args = Array.from(data);
+  if (args.length < 2) {
+    console.error('Invalid targeting rule. Require at least two buckets.')
```

Comment:
Same comment about the logger.

Reply:
I addressed this in the evaluator end. But adding this to targeting handlers is not that straightforward as we have to deal with method signatures of json logic engine